jerry-core/parser/js/js-parser-module.c

```diff
@@ -427,19 +432,30 @@ parser_module_parse_import_clause (parser_context_t *context_p) /**< parser cont
       if (context_p->token.type != LEXER_LITERAL
           || context_p->token.lit_location.type != LEXER_IDENT_LITERAL)
       {
-        ecma_deref_ecma_string (local_name_p);
-        ecma_deref_ecma_string (import_name_p);
         parser_raise_error (context_p, PARSER_ERR_IDENTIFIER_EXPECTED);
       }
 
       lexer_construct_literal_object (context_p, &context_p->token.lit_location, LEXER_IDENT_LITERAL);
 
-      ecma_deref_ecma_string (local_name_p);
-      local_name_p = ecma_new_ecma_string_from_utf8 (context_p->lit_object.literal_p->u.char_p,
-                                                     context_p->lit_object.literal_p->prop.length);
+      import_name_index = context_p->lit_object.index;
+
       lexer_next_token (context_p);
     }
 
+    lexer_literal_t *literal_p = PARSER_GET_LITERAL (local_name_index);
+    local_name_p = ecma_new_ecma_string_from_utf8 (literal_p->u.char_p, literal_p->prop.length);
+    import_name_p = local_name_p;
+
+    if (import_name_index != PARSER_MAXIMUM_NUMBER_OF_LITERALS)
+    {
+      lexer_literal_t *as_literal_p = PARSER_GET_LITERAL (import_name_index);
+      local_name_p = ecma_new_ecma_string_from_utf8 (as_literal_p->u.char_p, as_literal_p->prop.length);
+    }
+    else
+    {
```

Comment:
Please also move the `local_name_p = import_name_p;` assignment into this block, similarly to the export case.